### PR TITLE
File type detection fails when multiple files are imported at once

### LINKED ISSUE #6772: File type detection fails when multiple files are imported at once
The heuristic to detect file types seems to have a weird behavior when working on multiple files, even though those files have a consistent file type which OpenRefine is able to detect correctly when they are imported individually

### To Reproduce

Create two text files `test.md` and `test2.md`, both containing the following content:
```
## Markdown test

This is a [test](https://gnu.org) test
```

2. Import both of those files at the same time (by selecting them both in the same folder)

### Current Results

The files are incorrectly detected as Excel files, which obviously fails.

### Expected Behavior

They should be detected as text files (which indeed happens when selecting only one file)

### Screenshots

image

### Versions
 - Operating System: Debian testing
 - JRE or JDK Version: 19
 - OpenRefine: 3.9-SNAPSHOT
 
### Additional context
Observed at Wikimania 2024

### PR DESCRIPTION
**Pull request by sunilnatraj**
File format info was not getting set when multiple files are uploaded.

Fixes #6772 

Changes proposed in this pull request:
- Set format in the import flow for multiple files
- Identify format based on the file, added guessBetterFormat() method
- Tested with multiple file types the format is getting set correctly

### CODE REVIEW

**Comment by sunilnatraj** [comment]
@wetneb Please review

**Review by wetneb — COMMENTED** [review]
Thank you for tackling this! I haven't had a close look at the code but for a change like this I think it would be worth having some tests, would you be up for adding some?

**Comment by sunilnatraj** [comment]
@wetneb I will check and update the test cases

File: main/src/com/google/refine/importing/ImportingUtilities.java [hunk 1]
```diff
@@ -1040,6 +1041,32 @@ static String guessBetterFormat(ImportingJob job, ArrayNode fileRecords, String
         return bestFormat;
     }
 
+    static String guessBetterFormat(File file, String fileEncoding, String bestFormat) {
```
Comment (tfmorris) [inline main/src/com/google/refine/importing/ImportingUtilities.java:1044]: It looks like the bulk of the code in this method is duplicated from the method immediately above. 
Can things be refactored to reduce/eliminate the duplication?

**Review by tfmorris — CHANGES_REQUESTED** [review]
Thanks for the contribution! In addition to @wetneb 's suggestion for test coverage, it looks like you might be able to reduce them amount of duplicated code in line with the DRY (Don't Repeat Yourself) principle.

File: main/src/com/google/refine/importing/ImportingUtilities.java (hunk 1 above)
Comment (sunilnatraj) [reply main/src/com/google/refine/importing/ImportingUtilities.java:1025]: @tfmorris I have addressed this.

**Comment by sunilnatraj** [comment]
@tfmorris @wetneb Added test cases and removed redundant code. Please review

**Comment by tfmorris** [comment]
Thanks for the updates! The code looks good and my functional testing was good. I'm not sure the file selection code is working as intended, but I think that's a separate problem.

The test failure, however, appears real. I've re-run the tests a couple of times and the failure has persisted, so I don't think it's just a flaky test.

**Comment by tfmorris** [comment]
Let's have another look at the test failure here

**Comment by tfmorris** [comment]
I had a quick look at the failing tests and it looks like they're all failing for the same reason - it's guessing that a CSV file is just a Line oriented text file instead of the more specific CSV file type.

You should be able to reproduce the problem by importing the attached file from the test. It should be guessed as a CSV file.

[food.mini.csv](https://github.com/user-attachments/files/16723762/food.mini.csv)

**Comment by sunilnatraj** [comment]
@tfmorris I have addressed the issue; can you please review the PR

**Review by tfmorris — APPROVED** [review]
Looks great! Thanks for writing the tests too.

It looks like perhaps you merged `master` into your branch and things got messed up, so I rebased things onto master. Generally I think `rebase` is better for syncing things up than `merge` but it's somewhat a matter of preference.

Two other minor things I fixed up:
- camel case test names
- removed try/except block and just let the test framework fail on any errors.

**Comment by tfmorris** [comment]
And congratulations on your first pull request merged!

**Comment by sunilnatraj** [comment]
Thank you @tfmorris

**Pull request merged**